src/device.ts

```diff
@@ -396,6 +396,25 @@ export class Device extends vscode.Disposable {
         });
     }
 
+    public getSystemInfo(): Promise<string> {
+        return new Promise((resolve, reject) => {
+            this.exec('ev3dev-sysinfo').then(conn => {
+                let sysinfoString = '';
+                conn.stdout.on('data', buffer => sysinfoString += buffer.toString());
+                conn.stdout.on('error', (err) => {
+                    conn.close();
+                    conn.stdout.removeAllListeners('end');
```

Comment:
probably don't need to bother removing listeners since conn is not a long lived object.

(I think... I don't really have a good grasp on js memory management yet)

Reply:
I think the goal here is to avoid the fact that `end` is called despite there being an error.

Reply:
Once a promise is resolved/rejected, it ignores any further calls to resolve or reject.

Reply:
That's true, but we currently do more than just that. I haven't yet validated the internal behavior WRT calling `close`.